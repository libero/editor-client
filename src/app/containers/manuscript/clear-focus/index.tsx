import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';
import * as manuscriptEditorAction from 'app/actions/manuscript-editor.actions';

export const ClearFocus: React.FC<{ className?: string }> = ({ children, className }) => {
  const dispatch = useDispatch();
  const handleClick = useCallback(() => {
    dispatch(manuscriptEditorAction.removeFocusAction());
  }, [dispatch]);

  return (
    <div className={className} onClick={handleClick}>
      {children}
    </div>
  );
};
