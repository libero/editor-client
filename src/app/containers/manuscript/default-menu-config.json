{
  "button-group": [
    "save",
    "undo",
    "redo",
    "pdf"
  ],
  "menus": [
    "headings",
    "typography",
    "insert"
  ],
  "custom-buttons": []
}