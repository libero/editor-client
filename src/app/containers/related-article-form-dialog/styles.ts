import { makeStyles } from '@material-ui/core/styles';

export const useRelatedArticleStyles = makeStyles((theme) => ({
  root: {},

  inputField: {
    width: 500,
    marginBottom: theme.spacing(3)
  },

  buttonPanel: {
    display: 'flex',
    justifyContent: 'flex-end',
    marginTop: theme.spacing(1),
    '& > *:last-child': {
      marginLeft: theme.spacing(2)
    }
  },

  selectLabel: {
    background: theme.palette.background.paper,
    padding: theme.spacing(0, 0.5)
  },

  articleTypeGreyed: {
    color: theme.palette.text.disabled
  },

  spacer: {
    flexGrow: 1
  }
}));
