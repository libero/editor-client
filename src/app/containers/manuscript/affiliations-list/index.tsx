import React, { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { IconButton } from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';

import { SectionContainer } from 'app/components/section-container';
import { ActionButton } from 'app/components/action-button';
import { getAffiliations } from 'app/selectors/manuscript.selectors';
import { Affiliation } from 'app/models/affiliation';
import { useAffiliationStyles } from './styles';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import { ConnectedAffiliationFormDialog } from 'app/containers/affiliation-form-dialog';
import { ComponentWithId } from 'app/types/utility.types';

export const AffiliationsList: React.FC<ComponentWithId> = ({ id }) => {
  const classes = useAffiliationStyles();
  const affiliations = useSelector(getAffiliations);
  const dispatch = useDispatch();

  const editAffiliation = useCallback(
    (aff: Affiliation) => {
      dispatch(
        manuscriptEditorActions.showModalDialog({
          component: ConnectedAffiliationFormDialog,
          props: { affiliation: aff },
          title: 'Edit Affiliation'
        })
      );
    },
    [dispatch]
  );

  const addAffiliation = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: ConnectedAffiliationFormDialog,
        props: {},
        title: 'Add Affiliation'
      })
    );
  }, [dispatch]);

  const renderAffiliation = (aff: Affiliation): React.ReactNode => (
    <li key={aff.id} className={classes.listItem}>
      <div className={classes.orderLabel}>{affiliations.length > 1 ? `(${aff.label})` : undefined}</div>
      <div className={classes.affiliationInfo}>{aff.getDisplayName()}</div>
      <IconButton classes={{ root: classes.editButton }} onClick={editAffiliation.bind(null, aff)}>
        <EditIcon fontSize="small" />
      </IconButton>
    </li>
  );

  return (
    <section>
      <SectionContainer label="Affiliations" id={id}>
        <ul className={classes.root}>{affiliations.map(renderAffiliation)}</ul>
      </SectionContainer>
      <ActionButton
        title="Add Affiliation"
        variant="addEntity"
        onClick={addAffiliation}
        className={classes.addButton}
      />
    </section>
  );
};
