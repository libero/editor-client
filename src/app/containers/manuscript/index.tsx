import React from 'react';
import { useSelector } from 'react-redux';
import { Backdrop, CircularProgress } from '@material-ui/core';

import { isManuscriptLoaded } from '../../selectors/manuscript.selectors';
import './styles.scss';
import { ManuscriptToolbar } from './manuscript-toolbar';
import { ManuscriptEditor } from './manuscript-editor';

const renderBackdrop = (): JSX.Element => (
  <Backdrop open={true}>
    <CircularProgress color="inherit" />
  </Backdrop>
);

export const ManuscriptContainer: React.FC = () => {
  const isLoaded = useSelector(isManuscriptLoaded);
  const renderContent = (): JSX.Element => (
    <React.Fragment>
      <ManuscriptToolbar />
      <ManuscriptEditor />
    </React.Fragment>
  );

  return isLoaded ? renderContent() : renderBackdrop();
};
