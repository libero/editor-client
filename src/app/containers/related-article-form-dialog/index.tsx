import React, { useCallback, ChangeEvent, useState } from 'react';
import { TextField, FormControl, InputLabel, MenuItem, Select } from '@material-ui/core';
import { useDispatch } from 'react-redux';
import classNames from 'classnames';

import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import * as manuscriptActions from 'app/actions/manuscript.actions';
import { ActionButton } from 'app/components/action-button';
import { createNewRelatedArticle, RelatedArticle } from 'app/models/related-article';
import { useRelatedArticleStyles } from 'app/containers/related-article-form-dialog/styles';
import { PromptDialog } from 'app/components/prompt-dialog';

const labelProps = { shrink: true };

interface RelatedArticleFormDialogProps {
  article?: RelatedArticle;
}

const renderConfirmDialog = (title: string, msg: string, onAccept: () => void, onReject: () => void) => {
  return (
    <PromptDialog
      title={title}
      message={msg}
      isOpen={true}
      onAccept={onAccept}
      onReject={onReject}
      acceptLabel="Delete"
      rejectLabel="Cancel"
      acceptVariant="containedWarning"
      rejectVariant="secondaryOutlined"
    />
  );
};

export const RelatedArticleFormDialog: React.FC<RelatedArticleFormDialogProps> = ({ article }) => {
  const [userArticle, setUserArticle] = useState(article || createNewRelatedArticle());
  const [isConfirmShown, setConfirmSnow] = useState<boolean>(false);
  const isNewArticle = !article;
  const classes = useRelatedArticleStyles();
  const dispatch = useDispatch();

  const closeDialog = useCallback(() => {
    dispatch(manuscriptEditorActions.hideModalDialog());
  }, [dispatch]);

  const handleAccept = useCallback(() => {
    setConfirmSnow(false);
    dispatch(manuscriptActions.deleteRelatedArticleAction(userArticle));
    closeDialog();
  }, [setConfirmSnow, userArticle, closeDialog, dispatch]);

  const handleReject = useCallback(() => {
    setConfirmSnow(false);
  }, [setConfirmSnow]);

  const handleFormChange = useCallback(
    (event: ChangeEvent<{ name?: string; value: string }>) => {
      const fieldName = event.target['name'];
      const newValue = event.target['value'];
      setUserArticle({
        ...userArticle,
        [fieldName]: newValue
      });
    },
    [userArticle, setUserArticle]
  );

  const handleDone = useCallback(() => {
    if (isNewArticle) {
      dispatch(manuscriptActions.addRelatedArticleAction(userArticle));
    } else {
      dispatch(manuscriptActions.updateRelatedArticleAction(userArticle));
    }

    closeDialog();
  }, [closeDialog, userArticle, dispatch, isNewArticle]);

  const handleDelete = useCallback(() => {
    setConfirmSnow(true);
  }, [closeDialog, dispatch, userArticle]);

  return (
    <section className={classes.root}>
      <TextField
        fullWidth
        name="href"
        label="Article DOI"
        classes={{ root: classes.inputField }}
        InputLabelProps={labelProps}
        variant="outlined"
        multiline
        value={userArticle.href}
        onChange={handleFormChange}
      />
      <FormControl variant="outlined" className={classes.inputField} fullWidth>
        <InputLabel className={classes.selectLabel} shrink id="related-article-type-label">
          Article type
        </InputLabel>
        <Select
          name="articleType"
          displayEmpty
          className={classNames({
            [classes.articleTypeGreyed]: !userArticle.articleType
          })}
          labelId="related-article-type-label"
          value={userArticle.articleType}
          onChange={handleFormChange}
          label="Article type"
        >
          <MenuItem value={''}>Please select</MenuItem>
          <MenuItem value={'article-reference'}>Article reference</MenuItem>
          <MenuItem value={'commentary'}>Commentary</MenuItem>
          <MenuItem value={'commentary-article'}>Commentary article</MenuItem>
          <MenuItem value={'corrected-article'}>Corrected article</MenuItem>
          <MenuItem value={'retracted-article'}>Retracted article</MenuItem>
        </Select>
      </FormControl>
      <div className={classes.buttonPanel}>
        {!isNewArticle ? <ActionButton variant="outlinedWarning" onClick={handleDelete} title="Delete" /> : undefined}
        <div aria-hidden={true} className={classes.spacer}></div>
        <ActionButton variant="secondaryOutlined" onClick={closeDialog} title="Cancel" />
        <ActionButton variant="primaryContained" onClick={handleDone} title="Done" />
      </div>
      {isConfirmShown
        ? renderConfirmDialog(
            'You are deleting a related article',
            'Are you sure you want to proceed?',
            handleAccept,
            handleReject
          )
        : undefined}
    </section>
  );
};
