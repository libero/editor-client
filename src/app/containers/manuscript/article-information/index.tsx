import React, { useCallback } from 'react';
import { IconButton } from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';
import { useSelector, useDispatch } from 'react-redux';

import { SectionContainer } from 'app/components/section-container';
import { getArticleInformation } from 'app/selectors/manuscript.selectors';
import { useArticleInformationStyles } from 'app/containers/manuscript/article-information/styles';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import { ArticleInfoFormDialog } from 'app/containers/article-info-form-dialog';

export const ArticleInformation: React.FC<{}> = () => {
  const articleInfo = useSelector(getArticleInformation);
  const dispatch = useDispatch();
  const editArticleInfo = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: ArticleInfoFormDialog,
        title: 'Article Information'
      })
    );
  }, [dispatch]);

  const classes = useArticleInformationStyles();
  return (
    <SectionContainer label="Article information" className={classes.sectionContainer}>
      <section className={classes.infoSection}>
        <div>
          <strong> Article DOI: </strong> {articleInfo.articleDOI}
        </div>
        <div>
          <strong> Publisher ID: </strong> {articleInfo.publisherId}
        </div>
      </section>
      <IconButton classes={{ root: classes.editButton }} onClick={editArticleInfo}>
        <EditIcon fontSize="small" />
      </IconButton>
    </SectionContainer>
  );
};
