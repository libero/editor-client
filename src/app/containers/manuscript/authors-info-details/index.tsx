import React, { useCallback } from 'react';
import { IconButton } from '@material-ui/core';
import { useSelector, useDispatch } from 'react-redux';
import EditIcon from '@material-ui/icons/Edit';
import { isBoolean, isEqual } from 'lodash';
import Interweave from 'interweave';

import { SectionContainer } from 'app/components/section-container';
import { getAuthorAffiliations, getAuthors } from 'app/selectors/manuscript.selectors';
import { Person } from 'app/models/person';
import {
  useAuthorDetailStyles,
  useAuthorsDetailsListStyles
} from 'app/containers/manuscript/authors-info-details/styles';
import { ActionButton } from 'app/components/action-button';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import { AuthorFormDialog } from 'app/containers/author-form-dialog';
import { stringifyEditorState } from 'app/utils/view.utils';
import { OrcidIcon } from 'app/assets/icons';
import { ComponentWithId } from 'app/types/utility.types';

export const AuthorsInfoDetails: React.FC<ComponentWithId> = ({ id }) => {
  const authors = useSelector(getAuthors);
  const classes = useAuthorsDetailsListStyles();
  const dispatch = useDispatch();

  const addAuthor = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: AuthorFormDialog,
        title: 'Add Author'
      })
    );
  }, [dispatch]);

  return (
    <section>
      <SectionContainer label="Author Information" id={id}>
        {authors.map((author) => (
          <AuthorInfo key={author.id} author={author} />
        ))}
      </SectionContainer>
      <ActionButton variant="addEntity" title="Add Author" onClick={addAuthor} className={classes.addAuthorButton} />
    </section>
  );
};

interface AuthorInfoProps {
  author: Person;
}

const AuthorInfo: React.FC<AuthorInfoProps> = React.memo(({ author }) => {
  const classes = useAuthorDetailStyles();
  const dispatch = useDispatch();
  const affiliations = useSelector(getAuthorAffiliations)(author);

  const editAuthor = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: AuthorFormDialog,
        props: { author },
        title: 'Edit Author'
      })
    );
  }, [author, dispatch]);

  return (
    <section className={classes.root}>
      <div className={classes.authorInfoContainer}>
        <div className={classes.authorInfoLine}>
          <strong> {author.getDisplayName()} </strong>
        </div>
        <div className={classes.authorInfoLine}>
          {author.isCorrespondingAuthor ? 'Corresponding Author: ' : undefined}
          {author.email}
        </div>
        <div className={classes.authorInfoLine}>
          {affiliations.map((aff) => (
            <div key={aff.id}>{aff.getDisplayName()}</div>
          ))}
        </div>
        <div className={classes.authorInfoLine}>
          {author.isAuthenticated ? <OrcidIcon className={classes.orcidIcon} /> : undefined}{' '}
          <a href={`https://orcid.org/${author.orcid}`} target="_blank" rel="noopener noreferrer">
            {author.orcid}
          </a>
        </div>
        {isBoolean(author.hasCompetingInterest) ? (
          <div className={classes.authorInfoLine}>
            {author.hasCompetingInterest
              ? `Competing interest: ${author.competingInterestStatement}`
              : 'No Competing interest declared'}
          </div>
        ) : undefined}
        {author.bio ? (
          <div className={classes.authorBio}>
            <Interweave content={stringifyEditorState(author.bio)} />
          </div>
        ) : undefined}
      </div>
      <IconButton classes={{ root: classes.editButton }} onClick={editAuthor}>
        <EditIcon fontSize="small" />
      </IconButton>
    </section>
  );
}, isEqual);
