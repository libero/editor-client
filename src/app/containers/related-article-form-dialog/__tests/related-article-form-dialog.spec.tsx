import React from 'react';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

import { givenState } from 'app/test-utils/reducer-test-helpers';
import * as manuscriptActions from 'app/actions/manuscript.actions';
import { RelatedArticleFormDialog } from 'app/containers/related-article-form-dialog/index';

jest.mock('@material-ui/core', () => ({
  Select: ({ children }) => <div data-cmp="Index">{children}</div>,
  MenuItem: ({ children }) => <div data-cmp="MenuItem">{children}</div>,
  FormControl: ({ children }) => <div data-cmp="FormControl">{children}</div>,
  InputLabel: ({ children }) => <div data-cmp="InputLabel">{children}</div>,
  TextField: () => <div data-cmp="TextField"></div>,
  Button: () => <div data-cmp="Button"></div>
}));

describe('Related Article Form Dialog', () => {
  const mockStore = configureMockStore([]);
  let mockState;

  beforeEach(() => {
    mockState = givenState({
      relatedArticles: [
        {
          id: 'ra1',
          articleType: 'commentary-article',
          href: '10.7554/eLife.42697'
        },
        {
          id: 'ra2',
          articleType: 'commentary-article',
          href: '10.7554/eLife.00067'
        }
      ]
    });
  });

  it('renders related article form', () => {
    const store = mockStore({
      manuscript: mockState
    });

    const wrapper = create(
      <Provider store={store}>
        <RelatedArticleFormDialog />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches update related article action', () => {
    const store = mockStore({ manuscript: mockState });
    jest.spyOn(store, 'dispatch');

    const wrapper = mount(
      <Provider store={store}>
        <RelatedArticleFormDialog article={mockState.data.present.relatedArticles[0]} />
      </Provider>
    );

    wrapper.find({ title: 'Done' }).prop('onClick')();
    expect(store.dispatch).toHaveBeenCalledWith(
      manuscriptActions.updateRelatedArticleAction(mockState.data.present.relatedArticles[0])
    );
  });

  it('dispatches add related article action', () => {
    const store = mockStore({ manuscript: mockState });
    jest.spyOn(store, 'dispatch');

    const wrapper = mount(
      <Provider store={store}>
        <RelatedArticleFormDialog />
      </Provider>
    );

    wrapper.find({ title: 'Done' }).prop('onClick')();
    expect(store.dispatch).toHaveBeenCalledWith(
      manuscriptActions.addRelatedArticleAction({
        articleType: '',
        href: '',
        id: expect.any(String)
      })
    );
  });
});
