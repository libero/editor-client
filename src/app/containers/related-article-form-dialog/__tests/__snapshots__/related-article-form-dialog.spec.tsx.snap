// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related Article Form Dialog renders related article form 1`] = `
<section
  className="makeStyles-root-1"
>
  <div
    data-cmp="TextField"
  />
  <div
    data-cmp="FormControl"
  >
    <div
      data-cmp="InputLabel"
    >
      Article type
    </div>
    <div
      data-cmp="Select"
    >
      <div
        data-cmp="MenuItem"
      >
        Please select
      </div>
      <div
        data-cmp="MenuItem"
      >
        Article reference
      </div>
      <div
        data-cmp="MenuItem"
      >
        Commentary
      </div>
      <div
        data-cmp="MenuItem"
      >
        Commentary article
      </div>
      <div
        data-cmp="MenuItem"
      >
        Corrected article
      </div>
      <div
        data-cmp="MenuItem"
      >
        Retracted article
      </div>
    </div>
  </div>
  <div
    className="makeStyles-buttonPanel-3"
  >
    <div
      aria-hidden={true}
      className="makeStyles-spacer-6"
    />
    <div
      data-cmp="Button"
    />
    <div
      data-cmp="Button"
    />
  </div>
</section>
`;
